File: python/02631_IntrProg/week006/letterFrequency.py
import numpy as np
 
def letterFrequency(filename):
    # Computes the frequency of each letter, a to z, case insensitive
    # in the text file (filename given as input)
 
    # Define alphabet
    alphabet = "abcdefghijklmnopqrstuvwxyz"
 
    # Read file into string
    filein = open(filename, "r")
    lines = filein.readlines()
    text = "".join(lines)
 
    # Convert to lower case
    text = text.lower()
 
    # Compute number of occurencies
    occur = np.zeros(26)
    for i in range(26):
        occur[i] = text.count(alphabet[i])
 
    # Compute frequencies in percent
    freq = occur / np.sum(occur) * 100
    return freq

File: MATLAB/02631_IntrProg/week006/letterFrequency.m
function freq = letterFrequency(filename)

% INPUT
% filename  :   A string that is the filename of a plain text file in the current working directory.
% OUTPUT
% freq      :   A vector of length 26 containing the frequency of occurrence of each of the 26 letters from a to z.
smalltxt = fileread(filename);
stLower = lower(smalltxt);
alpha = ['a':'z'];
count = zeros(1,length(alpha));
for i = 1:length(stLower)
    count = count + (stLower(i) == alpha);
end
tOccur = sum(count);
freq = (count/tOccur)*100;
end